### PR TITLE
[0-size Tensor No.202、209、220、225] Add 0-size Tensor support for mse_loss

### PR DESCRIPTION
**Pull request by co63oc**
### PR Category

Execute Infrastructure 

### PR Types

Improvements 

### Description

202 paddle.nn.functional.mse_loss
209 paddle.nn.functional.poisson_nll_loss
220 paddle.nn.functional.soft_margin_loss
225 paddle.nn.functional.square_error_cost

增加单测
soft_margin_loss 在report/0size_tensor_gpu/test_history/20250528/accuracy/error_log.log 没有配置和日志信息，应该是没有错误
其他算子GPU测试都通过
CPU测试 paddle.nn.functional.poisson_nll_loss paddle.nn.functional.mse_loss 不支持float16

### CODE REVIEW

**Comment by DanielSun11** [comment]
单测挂了

**Review by DanielSun11 — APPROVED** [review]
LGTM

**Review by wanghuancoder — APPROVED** [review]
